eclair-core/src/main/scala/fr/acinq/eclair/channel/ChannelTypes.scala

```diff
@@ -199,9 +202,39 @@ final case class LocalParams(nodeId: PublicKey,
                              maxAcceptedHtlcs: Int,
                              isFunder: Boolean,
                              defaultFinalScriptPubKey: ByteVector,
+                             localPaymentBasepoint: Option[PublicKey],
```

Comment:
There should be a check that `localPaymentBasepoint` can be defined iff option `static_remote_key` is set

Reply:
I've added the check in `Commitments` since we don't have the `channelVersion` down in `LocalParams`